lib/SPIRV/SPIRVToLLVMDbgTran.cpp

```diff
@@ -1037,15 +1037,18 @@ SPIRVToLLVMDbgTran::ParseChecksum(StringRef Text) {
   // Example of "Text" variable:
   // "SomeInfo//__CSK_MD5:7bb56387968a9caa6e9e35fff94eaf7b:OtherInfo"
   Optional<DIFile::ChecksumInfo<StringRef>> CS;
-  auto KindPos = Text.find(SPIRVDebug::ChecksumKindPrefx);
-  if (KindPos != StringRef::npos) {
-    auto ColonPos = Text.find(":", KindPos);
-    KindPos += string("//__").size();
-    auto KindStr = Text.substr(KindPos, ColonPos - KindPos);
-    auto Checksum = Text.substr(ColonPos).ltrim(':');
-    if (auto Kind = DIFile::getChecksumKind(KindStr)) {
-      size_t ChecksumEndPos = Checksum.find_if_not(llvm::isHexDigit);
-      CS.emplace(Kind.getValue(), Checksum.substr(0, ChecksumEndPos));
+  if (Text.size() > SPIRVDebug::ChecksumKindPrefx.size() &&
```

Comment:
Just a nit: may we return earlier if first two conditional checks are not met? It'd reduce nesting of code blocks.